internal/project/checkerpool.go

```diff
@@ -75,6 +78,23 @@ func (p *CheckerPool) GetCheckerForFile(ctx context.Context, file *ast.SourceFil
 	return checker, p.createRelease(requestID, index, checker)
 }
 
+// GetCheckerForFileExclusive is the same as GetCheckerForFile but also locks a mutex associated with the checker.
+// Call `done` to free the lock.
+func (p *CheckerPool) GetCheckerForFileExclusive(ctx context.Context, file *ast.SourceFile) (*checker.Checker, func()) {
+	c, done := p.GetCheckerForFile(ctx, file)
```

Comment:
This implementation of checker pool can just leave this method unimplemented if it's only going to be used by incremental build. I have a cleanup/simplification of CheckerPool on the backburner so minimizing the surface area would be good.